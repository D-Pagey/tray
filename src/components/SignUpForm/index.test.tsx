import React from 'react';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SignUpForm, SignUpFormTypes } from '.';

const props: SignUpFormTypes = {
    onSubmit: () => null
};

describe('SignUpForm component', () => {
    it('should render', () => {
        const { container } = render(<SignUpForm {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onSubmit with correct values', async () => {
        const name = 'Dan';
        const role = 'Software Engineer';
        const email = 'dan@gmail.com';
        const password = 'Tray10SecurePassword';
        const onSubmit = jest.fn();

        const { getByTestId } = render(<SignUpForm {...props} onSubmit={onSubmit} />);

        userEvent.type(getByTestId('signUpNameInput'), name);

        userEvent.type(getByTestId('signUpRoleInput'), role);

        userEvent.type(getByTestId('signUpEmailInput'), email);

        userEvent.type(getByTestId('signUpPasswordInput'), password);

        userEvent.click(getByTestId('signUpSubmitButton'));

        await waitFor(() => expect(onSubmit).toHaveBeenCalledWith({ name, role, email, password }));
    });

    it.todo('should show errors for each required field (use it.each)');
});
