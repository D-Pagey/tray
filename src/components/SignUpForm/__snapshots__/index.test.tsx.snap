// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component should render 1`] = `
<div
  class="sc-AxirZ jfPcbx"
>
  <form
    action="#"
    class="sc-AxiKw iSgsng"
  >
    <label
      class="sc-AxhUy jBwtVz"
      for="name"
    >
      Name:
      <input
        class="sc-AxhCb bMSIvc"
        data-testid="signUpNameInput"
        id="name"
        name="name"
        type="text"
        value=""
      />
    </label>
    <label
      class="sc-AxhUy jBwtVz"
      for="role"
    >
      Role:
      <input
        class="sc-AxhCb bMSIvc"
        data-testid="signUpRoleInput"
        id="role"
        name="role"
        type="text"
        value=""
      />
    </label>
    <label
      class="sc-AxhUy jBwtVz"
      for="email"
    >
      Email:
      <input
        class="sc-AxhCb bMSIvc"
        data-testid="signUpEmailInput"
        id="email"
        name="email"
        type="email"
        value=""
      />
    </label>
    <label
      class="sc-AxhUy jBwtVz"
      for="password"
    >
      Password:
      <input
        class="sc-AxhCb bMSIvc"
        data-testid="signUpPasswordInput"
        id="password"
        name="password"
        type="password"
        value=""
      />
    </label>
    <button
      class="sc-AxjAm bRcXEK"
      data-testid="signUpSubmitButton"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
